array-filter/filter.js

```diff
@@ -0,0 +1,78 @@
+const numbersArray = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+const namesArray = [
+  'Ada',
+  'Hedy',
+  'Jean',
+  'Grace',
+  'Evelyn',
+  'Joan',
+  'Elizabeth',
+  'Janese',
+  'Donna'
+];
+
+function filter(array, predicate) {
```

Comment:
Here `predicate` should be a callback function. Inside of your `filter` function you should loop over the array and call the `predicate` callback function for each item in the array, using the result of calling the function to determine what should or shouldn't be added to the array that will be returned from `filter`.